Fix smartmontools ebuild warning -

jamesbroadhead opened the pull request:
- fix name of command to update the drivedb (make a variable, so that it is common between usages)
- update warning about updating the drivedb, so that it notices the useflag
- fix description so that it's short enough (from the smartmontools website) -- repoman
- quote some variables -- shellcheck
- update the copy of the warning 

Gentoo bug: https://bugs.gentoo.org/show_bug.cgi?id=587372

Polynomial-C commented on sys-apps/smartmontools/smartmontools-6.4.ebuild:
```diff
@@ -49,9 +49,9 @@ src_configure() {
-		$(systemd_with_unitdir)
+		"$(use_with caps libcap-ng)" \
+		"$(use_with selinux)" \
+		"$(systemd_with_unitdir)"
```
These quoting changes are completely useless. Please remove.

Polynomial-C commented on sys-apps/smartmontools/smartmontools-6.5.ebuild:
```diff
@@ -84,16 +86,21 @@ pkg_postinst() {
+			ewarn "database by: "
+
+			if ! use update_drivedb ; then
+				ewarn "re-merging ${PN} with USE='update_drivedb, then"
```
Please fix the `USE=` string. The way it is given now it can confuse the user as it is incomplete.

jamesbroadhead replied:
you mean $(USE="update_drivedb" emerge smartmontools) ? 
I've never understood why ebuilds print this, as setting them like this is impermanent.

Polynomial-C replied:
now it is 

`ewarn "re-merging ${PN} with USE='update_drivedb, then"`

which should rather be:

`ewarn "re-merging ${PN} with USE='update_drivedb', then"`

robbat2 commented on sys-apps/smartmontools/smartmontools-9999.ebuild:
```diff
@@ -84,16 +86,21 @@ pkg_postinst() {
+			ewarn "running the following command as root:"
 			ewarn ""
-			ewarn "/usr/sbin/update-smart-drivedb"
+			ewarn "${UPDATE_DRIVEDB}"
```
This prints `.../files/smartmontools-update-drivedb` instead of the installed path, which I don't think you intended.

robbat2 commented on sys-apps/smartmontools/smartmontools-6.5.ebuild:
```diff
@@ -84,16 +86,21 @@ pkg_postinst() {
+			ewarn "running the following command as root:"
 			ewarn ""
-			ewarn "/usr/sbin/update-smart-drivedb"
+			ewarn "${UPDATE_DRIVEDB}"
```
This prints `.../files/smartmontools-update-drivedb` instead of the installed path, which I don't think you intended.

robbat2 commented on sys-apps/smartmontools/smartmontools-6.4-r1.ebuild:
```diff
@@ -85,14 +87,14 @@ pkg_postinst() {
 			ewarn "database by running the following command as root:"
 			ewarn ""
-			ewarn "/usr/sbin/update-smart-drivedb"
+			ewarn "${UPDATE_DRIVEDB}"
```
This prints `.../files/smartmontools-update-drivedb` instead of the installed path, which I don't think you intended.

vapier commented:
the commit message style here is incorrect.  every commit should be prefixed with `sys-apps/smartmontools: `.

should update our CI checks to cover that ...

vapier commented on sys-apps/smartmontools/smartmontools-6.4.ebuild:
```diff
@@ -49,9 +49,9 @@ src_configure() {
 	econf \
 		--docdir="${EPREFIX}/usr/share/doc/${PF}" \
 		--with-initscriptdir="${EPREFIX}/etc/init.d" \
-		"$(use_with caps libcap-ng)" \
```
you added these errors in a previous commit.  please revert/squash so these are never a problem.

vapier commented on sys-apps/smartmontools/smartmontools-6.5.ebuild:
```diff
@@ -91,7 +91,7 @@ pkg_postinst() {
 			ewarn "database by: "
 
 			if ! use update_drivedb ; then
-				ewarn "re-merging ${PN} with USE='update_drivedb, then"
```
you added these errors in a previous commit.  please revert/squash so these are never a problem.

vapier commented on sys-apps/smartmontools/smartmontools-9999.ebuild:
```diff
@@ -91,7 +91,7 @@ pkg_postinst() {
 			ewarn "database by: "
 
 			if ! use update_drivedb ; then
-				ewarn "re-merging ${PN} with USE='update_drivedb, then"
```
you added these errors in a previous commit.  please revert/squash so these are never a problem.

jamesbroadhead commented:
updated - thanks for the reviews. 

@vapier I got an email saying you'd left a comment asking for a squash-commit - can't find it in here though? Squashed anyway

jamesbroadhead commented:
@vapier ok to merge?

monsieurp commented:
@gentoo/base-system can we move on with this PR?

monsieurp commented:
@robbat2 can you OK it?

Polynomial-C commented on sys-apps/smartmontools/smartmontools-6.4-r1.ebuild:
```diff
@@ -85,14 +87,14 @@ pkg_postinst() {
 			ewarn "database by running the following command as root:"
 			ewarn ""
-			ewarn "/usr/sbin/update-smart-drivedb"
+			ewarn "/usr/sbin/${UPDATE_DRIVEDB}"
```
This is wrong. The tool is called `update-smart-drivedb` and not `smartmontools-update-drivedb`. Please remove this change!

Polynomial-C commented on sys-apps/smartmontools/smartmontools-6.5.ebuild:
```diff
@@ -68,13 +70,13 @@ src_install() {
 
 			exeinto /etc/cron.monthly
-			doexe "${FILESDIR}"/${PN}-update-drivedb
+			doexe "${FILESDIR}/${UPDATE_DRIVEDB}"
```
Why are you trying to install this cron script with different name but did not either

- change the file's name in FILESDIR

or

- use `newexe` to rename the file on installation
Please fix!

Polynomial-C commented on sys-apps/smartmontools/smartmontools-6.5.ebuild:
```diff
@@ -84,16 +86,21 @@ pkg_postinst() {
+			ewarn "running the following command as root:"
 			ewarn ""
-			ewarn "/usr/sbin/update-smart-drivedb"
+			ewarn "/usr/sbin/${UPDATE_DRIVEDB}"
```
This is wrong. The tool is called update-smart-drivedb and not smartmontools-update-drivedb. Please remove this change!

Polynomial-C commented on sys-apps/smartmontools/smartmontools-9999.ebuild:
```diff
@@ -84,16 +86,21 @@ pkg_postinst() {
+			ewarn "running the following command as root:"
 			ewarn ""
-			ewarn "/usr/sbin/update-smart-drivedb"
+			ewarn "/usr/sbin/${UPDATE_DRIVEDB}"
```
This is wrong. The tool is called update-smart-drivedb and not smartmontools-update-drivedb. Please remove this change!

Polynomial-C commented on sys-apps/smartmontools/smartmontools-6.4-r1.ebuild:
```diff
@@ -72,11 +74,11 @@ src_install() {
 
 		exeinto /etc/cron.monthly
-		doexe "${FILESDIR}"/${PN}-update-drivedb
+		doexe "${FILESDIR}/${UPDATE_DRIVEDB}"
```
Why are you trying to install this cron script with different name but did not either

- change the file's name in FILESDIR

or

- use `newexe` to rename the file on installation
Please fix!

Polynomial-C commented on sys-apps/smartmontools/smartmontools-9999.ebuild:
```diff
@@ -68,13 +70,13 @@ src_install() {
 
 			exeinto /etc/cron.monthly
-			doexe "${FILESDIR}"/${PN}-update-drivedb
+			doexe "${FILESDIR}/${UPDATE_DRIVEDB}"
```
Why are you trying to install this cron script with different name but did not either

- change the file's name in FILESDIR

or

- use `newexe` to rename the file on installation
Please fix!

jamesbroadhead commented:
@Polynomial-C yep, I went too far trying to consolidate. In retrospect, having the cronjob prefixed with the package name is a good idea & the update-* script name follows the existing convention. Removed those changes.
